core/src/main.cpp

```diff
@@ -676,37 +676,51 @@ auto set_log_level(const json& _config) -> void
 
 auto init_tls(const json& _config) -> void
 {
-	const auto set_env_var_string =
-		[&](const auto& _json_ptr_path, const char* _env_var, const char* _default_value = "") {
-			using json_ptr = json::json_pointer;
-
-			if (const auto v = _config.value(json_ptr{_json_ptr_path}, _default_value); !v.empty()) {
+	// The iRODS connection libraries do not provide a clean way for developers to easily
+	// configure TLS without relying on an irods_environment.json file. All is not lost
+	// though. Turns out the iRODS libraries do inspect environment variables. This gives
+	// the HTTP API a way to hook into the connection logic and support TLS through its
+	// own configuration file. Hence, the following environment-based lambda functions.
+
+	const auto set_env_string =
+		[&](const auto& _tls_prop, const char* _env_var, const char* _default_value = "") {
+			const auto element_path = fmt::format("/irods_client/tls/{}", _tls_prop);
```

Comment:
this hard-coded irods_client/tls means that **other** environment variables are not possible...

should we pass two parameters to set_env_string and set_env_int?

no real use case yet - just thinking out loud.    ignore with confidence...